loader/loader.c

```diff
@@ -6043,6 +6031,8 @@ VkResult loader_create_instance_chain(const VkInstanceCreateInfo *pCreateInfo, c
 
             chain_info.u.pLayerInfo = &layer_instance_link_info[activated_layers];
 
+            activated_layer_names[activated_layers] = strdup(layer_prop->info.layerName);
```

Comment:
Looks like this is leaked? I think you'll either need to "borrow" the string (if the lifetime of `layer_prop->info.layerName` is guaranteed to be longer than `activated_layer_names`), or free the copied strings stored in `activated_layer_names` before it goes out of scope (otherwise).

If you're using `strdup()`, note that it can fail and return `NULL`.

Reply:
Ops, your are right. I guess I misinterpreted `alloca()` and forgot to free the actual elements.

I removed the `strdup()` because `layer_prop` is just a pointer to `dev->expanded_activated_layer_list.list[i]`, that does not change while we are in this "for" loop. So it is guaranteed to last longer than `activated_layer_names`.